### ISSUE TITLE
ImportError (GLIBCXX) when attempting to import segmentation module

### ISSUE THREAD

**Issue by jjl2** [scikit-image/scikit-image#6447]
## Way to reproduce
From the interpreter:
```python
import skimage.segmentation
```

## Version information
```python
# Paste the output of the following python commands
from __future__ import print_function
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print(f'scikit-image version: {skimage.__version__}')
import numpy; print(f'numpy version: {numpy.__version__}')
```

```python
3.10.4 (main, Mar 31 2022, 08:41:55) [GCC 7.5.0]
Linux-5.15.0-41-generic-x86_64-with-glibc2.35
scikit-image version: 0.19.2
numpy version: 1.22.3

```

**Comment by jjl2** [comment]
When I attempt to `import skimage.segmentation`, here is the last line of the traceback:

`ImportError: /home/john/miniconda3/envs/png16/lib/python3.10/site-packages/zmq/backend/cython/../../../../.././libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /home/john/miniconda3/envs/png16/lib/python3.10/site-packages/scipy/fft/_pocketfft/pypocketfft.cpython-310-x86_64-linux-gnu.so)`

Other skimage modules import successfully.

The output of the terminal command `strings /usr/lib/x86_64-linux-gnu/libstdc++.so.6 | grep GLIBCXX` states that I have many versions of GLIBCXX on my system, and GLIBCXX_3.4.30 is included.

**Comment by hmaarrfk** [comment]
can you provide the output from the following two commands, in full please:
```
conda info
conda list
```
thanks!

**Comment by jjl2** [comment]
@hmaarrfk Per your request:

```
$ conda info

     active environment : png16
    active env location : /home/john/miniconda3/envs/png16
            shell level : 2
       user config file : /home/john/.condarc
 populated config files : /home/john/.condarc
                          /home/john/miniconda3/envs/png16/.condarc
          conda version : 4.13.0
    conda-build version : not installed
         python version : 3.9.12.final.0
       virtual packages : __cuda=11.7=0
                          __linux=5.15.0=0
                          __glibc=2.35=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/john/miniconda3  (writable)
      conda av data dir : /home/john/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/john/miniconda3/pkgs
                          /home/john/.conda/pkgs
       envs directories : /home/john/miniconda3/envs
                          /home/john/.conda/envs
               platform : linux-64
             user-agent : conda/4.13.0 requests/2.27.1 CPython/3.9.12 Linux/5.15.0-41-generic ubuntu/22.04 glibc/2.35 solver/libmamba conda-libmamba-solver/22.6.0 libmambapy/0.24.0
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

```
$ conda list

# packages in environment at /home/john/miniconda3/envs/png16:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    conda-forge
_openmp_mutex             5.1                       1_gnu  
alabaster                 0.7.12                     py_0    conda-forge
arrow                     1.2.2              pyhd8ed1ab_0    conda-forge
astroid                   2.11.6          py310hff52083_0    conda-forge
atomicwrites              1.4.1              pyhd8ed1ab_0    conda-forge
attrs                     21.4.0             pyhd8ed1ab_0    conda-forge
autopep8                  1.6.0              pyhd8ed1ab_1    conda-forge
babel                     2.10.3             pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
beautifulsoup4            4.11.1             pyha770c72_0    conda-forge
binaryornot               0.4.4                      py_1    conda-forge
black                     22.6.0             pyhd8ed1ab_0    conda-forge
bleach                    5.0.1              pyhd8ed1ab_0    conda-forge
brotli                    1.0.9                h166bdaf_7    conda-forge
brotli-bin                1.0.9                h166bdaf_7    conda-forge
brotlipy                  0.7.0           py310h7f8727e_1002  
bzip2                     1.0.8                h7b6447c_0  
c-ares                    1.18.1               h7f8727e_0  
ca-certificates           2022.6.15            ha878542_0    conda-forge
certifi                   2022.6.15       py310hff52083_0    conda-forge
cffi                      1.15.0          py310hd667e15_1  
chardet                   5.0.0           py310hff52083_0    conda-forge
charset-normalizer        2.0.4              pyhd3eb1b0_0  
click                     8.1.3           py310hff52083_0    conda-forge
cloudpickle               2.1.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.5              pyhd8ed1ab_0    conda-forge
conda-package-handling    1.8.1           py310h7f8727e_0  
cookiecutter              2.1.1              pyh6c4a22f_0    conda-forge
cryptography              37.0.1          py310h9ce1e76_0  
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cytoolz                   0.11.2          py310h5764c6d_2    conda-forge
dask-core                 2022.7.0           pyhd8ed1ab_0    conda-forge
dataclasses               0.8                pyhc8e2a94_3    conda-forge
dbus                      1.13.6               h48d8840_2    conda-forge
debugpy                   1.6.0           py310hd8f1fbe_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
diff-match-patch          20200713           pyh9f0ad1d_0    conda-forge
dill                      0.3.5.1            pyhd8ed1ab_0    conda-forge
docutils                  0.18.1          py310hff52083_1    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
expat                     2.4.8                h27087fc_0    conda-forge
flake8                    4.0.1              pyhd8ed1ab_2    conda-forge
fontconfig                2.13.1               h6c09931_0  
fonttools                 4.33.3          py310h5764c6d_0    conda-forge
freetype                  2.10.4               h0708190_1    conda-forge
fsspec                    2022.5.0           pyhd8ed1ab_0    conda-forge
gettext                   0.19.8.1          h0b5b191_1005    conda-forge
giflib                    5.2.1                h36c2ea0_2    conda-forge
glib                      2.68.4               h9c3ff4c_0    conda-forge
glib-tools                2.68.4               h9c3ff4c_0    conda-forge
gst-plugins-base          1.14.0               h8213a91_2  
gstreamer                 1.14.0               h28cd5cc_2  
icu                       58.2                 he6710b0_3  
idna                      3.3                pyhd3eb1b0_0  
imagecodecs-lite          2019.12.3       py310hde88566_5    conda-forge
imageio                   2.19.3             pyhcf75d05_0    conda-forge
imagesize                 1.4.1              pyhd8ed1ab_0    conda-forge
importlib-metadata        4.11.3          py310h06a4308_0  
importlib_metadata        4.11.3               hd3eb1b0_0  
importlib_resources       5.8.0              pyhd8ed1ab_0    conda-forge
inflection                0.5.1              pyh9f0ad1d_0    conda-forge
intervaltree              3.0.2                      py_0    conda-forge
ipykernel                 6.15.1             pyh210e3f2_0    conda-forge
ipython                   7.33.0          py310hff52083_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
isort                     5.10.1             pyhd8ed1ab_0    conda-forge
jbig                      2.1               h7f98852_2003    conda-forge
jedi                      0.18.1          py310hff52083_1    conda-forge
jeepney                   0.8.0              pyhd8ed1ab_0    conda-forge
jellyfish                 0.9.0           py310h5764c6d_1    conda-forge
jinja2                    3.0.3              pyhd8ed1ab_0    conda-forge
jinja2-time               0.2.0              pyhd8ed1ab_3    conda-forge
jpeg                      9e                   h166bdaf_1    conda-forge
jsonschema                4.7.2              pyhd8ed1ab_0    conda-forge
jupyter_client            7.3.4              pyhd8ed1ab_0    conda-forge
jupyter_core              4.10.0          py310hff52083_0    conda-forge
jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
keyring                   23.6.0          py310hff52083_1    conda-forge
kiwisolver                1.4.2           py310hbf28c38_1    conda-forge
krb5                      1.19.2               hac12032_0  
lazy-object-proxy         1.7.1           py310h5764c6d_1    conda-forge
lcms2                     2.12                 hddcbb42_0    conda-forge
ld_impl_linux-64          2.38                 h1181459_1  
lerc                      2.2.1                h9c3ff4c_0    conda-forge
libarchive                3.5.2                h5de8990_0  
libblas                   3.9.0           15_linux64_openblas    conda-forge
libbrotlicommon           1.0.9                h166bdaf_7    conda-forge
libbrotlidec              1.0.9                h166bdaf_7    conda-forge
libbrotlienc              1.0.9                h166bdaf_7    conda-forge
libcblas                  3.9.0           15_linux64_openblas    conda-forge
libcurl                   7.82.0               h0b77cf5_0  
libdeflate                1.7                  h7f98852_5    conda-forge
libedit                   3.1.20210910         h7f8727e_0  
libev                     4.33                 h7f8727e_1  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 11.2.0               h1234567_1  
libgfortran-ng            12.1.0              h69a702a_16    conda-forge
libgfortran5              12.1.0              hdcd56e2_16    conda-forge
libglib                   2.68.4               h3e27bee_0    conda-forge
libgomp                   11.2.0               h1234567_1  
libiconv                  1.16                 h516909a_0    conda-forge
liblapack                 3.9.0           15_linux64_openblas    conda-forge
libmamba                  0.24.0               h1566912_0  
libmambapy                0.24.0          py310hd09550d_0  
libnghttp2                1.46.0               hce63b2e_0  
libopenblas               0.3.20          pthreads_h78a6416_0    conda-forge
libpng                    1.6.37               h21135ba_2    conda-forge
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libsolv                   0.7.20               h4ff587b_0  
libspatialindex           1.9.3                h9c3ff4c_4    conda-forge
libssh2                   1.10.0               h8f2d780_0  
libstdcxx-ng              11.2.0               h1234567_1  
libtiff                   4.3.0                hf544144_1    conda-forge
libuuid                   1.0.3                h7f8727e_2  
libwebp                   1.2.2                h3452ae3_0    conda-forge
libwebp-base              1.2.2                h7f98852_1    conda-forge
libxcb                    1.15                 h7f8727e_0  
libxml2                   2.9.14               h74e7548_0  
locket                    1.0.0              pyhd8ed1ab_0    conda-forge
lz4-c                     1.9.3                h295c915_1  
markupsafe                2.1.1           py310h5764c6d_1    conda-forge
matplotlib                3.5.2           py310hff52083_0    conda-forge
matplotlib-base           3.5.2           py310h5701ce4_0    conda-forge
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
mccabe                    0.6.1                      py_1    conda-forge
mistune                   0.8.4           py310h6acc77f_1005    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
mypy_extensions           0.4.3           py310hff52083_5    conda-forge
nbclient                  0.6.6              pyhd8ed1ab_0    conda-forge
nbconvert                 6.5.0              pyhd8ed1ab_0    conda-forge
nbconvert-core            6.5.0              pyhd8ed1ab_0    conda-forge
nbconvert-pandoc          6.5.0              pyhd8ed1ab_0    conda-forge
nbformat                  5.4.0              pyhd8ed1ab_0    conda-forge
ncurses                   6.3                  h5eee18b_3  
nest-asyncio              1.5.5              pyhd8ed1ab_0    conda-forge
networkx                  2.8.4              pyhd8ed1ab_0    conda-forge
numpy                     1.22.3          py310h4ef5377_2    conda-forge
numpydoc                  1.4.0              pyhd8ed1ab_0    conda-forge
openssl                   1.1.1q               h7f8727e_0  
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandoc                    2.18                 ha770c72_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
partd                     1.2.0              pyhd8ed1ab_0    conda-forge
pathspec                  0.9.0              pyhd8ed1ab_0    conda-forge
pcre                      8.45                 h9c3ff4c_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    9.0.1           py310h22f2fdc_0  
pip                       22.1.2          py310h06a4308_0  
platformdirs              2.5.1              pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0           py310hff52083_3    conda-forge
prompt-toolkit            3.0.30             pyha770c72_0    conda-forge
psutil                    5.9.1           py310h5764c6d_0    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pycodestyle               2.8.0              pyhd8ed1ab_0    conda-forge
pycosat                   0.6.3           py310h7f8727e_0  
pycparser                 2.21               pyhd3eb1b0_0  
pydocstyle                6.1.1              pyhd8ed1ab_0    conda-forge
pyflakes                  2.4.0              pyhd8ed1ab_0    conda-forge
pygments                  2.12.0             pyhd8ed1ab_0    conda-forge
pylint                    2.14.4             pyhd8ed1ab_0    conda-forge
pyls-spyder               0.4.0              pyhd8ed1ab_0    conda-forge
pyopenssl                 22.0.0             pyhd3eb1b0_0  
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
pyqt                      5.9.2           py310h295c915_6  
pyrsistent                0.18.1          py310h5764c6d_1    conda-forge
pysocks                   1.7.1           py310h06a4308_0  
python                    3.10.4               h12debd9_0  
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-fastjsonschema     2.15.3             pyhd8ed1ab_0    conda-forge
python-lsp-black          1.1.0              pyhd8ed1ab_0    conda-forge
python-lsp-jsonrpc        1.0.0              pyhd8ed1ab_0    conda-forge
python-lsp-server         1.3.3              pyhd8ed1ab_0    conda-forge
python-slugify            6.1.2              pyhd8ed1ab_0    conda-forge
python_abi                3.10                    2_cp310    conda-forge
pytoolconfig              1.2.1              pyhd8ed1ab_0    conda-forge
pytz                      2022.1             pyhd8ed1ab_0    conda-forge
pywavelets                1.3.0           py310hde88566_1    conda-forge
pyxdg                     0.28               pyhd8ed1ab_0    conda-forge
pyyaml                    6.0             py310h5764c6d_4    conda-forge
pyzmq                     23.0.0          py310h330234f_0    conda-forge
qdarkstyle                3.0.2              pyhd8ed1ab_0    conda-forge
qstylizer                 0.2.1              pyhd8ed1ab_0    conda-forge
qt                        5.9.7                h5867ecd_1  
qtawesome                 1.1.1              pyhd8ed1ab_0    conda-forge
qtconsole                 5.2.2              pyhd8ed1ab_1    conda-forge
qtconsole-base            5.2.2              pyhd8ed1ab_1    conda-forge
qtpy                      2.1.0              pyhd8ed1ab_0    conda-forge
readline                  8.1.2                h7f8727e_1  
reproc                    14.2.4               h295c915_1  
reproc-cpp                14.2.4               h295c915_1  
requests                  2.28.0          py310h06a4308_0  
rope                      1.2.0              pyhd8ed1ab_0    conda-forge
rtree                     1.0.0           py310hbdcdc62_1    conda-forge
ruamel_yaml               0.15.100        py310h7f8727e_0  
scikit-image              0.19.2          py310hb5077e9_0    conda-forge
scipy                     1.8.1           py310h7612f91_0    conda-forge
secretstorage             3.3.2           py310hff52083_1    conda-forge
setuptools                61.2.0          py310h06a4308_0  
sip                       4.19.13         py310h295c915_0  
six                       1.16.0             pyh6c4a22f_0    conda-forge
snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.3.1              pyhd8ed1ab_0    conda-forge
sphinx                    5.0.2              pyh6c4a22f_0    conda-forge
sphinxcontrib-applehelp   1.0.2                      py_0    conda-forge
sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
sphinxcontrib-htmlhelp    2.0.0              pyhd8ed1ab_0    conda-forge
sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
sphinxcontrib-qthelp      1.0.3                      py_0    conda-forge
sphinxcontrib-serializinghtml 1.1.5              pyhd8ed1ab_2    conda-forge
spyder                    5.2.2           py310hff52083_3    conda-forge
spyder-kernels            2.2.1           py310hff52083_1    conda-forge
sqlite                    3.38.5               hc218d9a_0  
text-unidecode            1.3                        py_0    conda-forge
textdistance              4.3.0              pyhd8ed1ab_0    conda-forge
three-merge               0.1.1              pyh9f0ad1d_0    conda-forge
tifffile                  2020.6.3                   py_0    conda-forge
tinycss2                  1.1.1              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h1ccaba5_0  
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
tomlkit                   0.11.1             pyha770c72_0    conda-forge
toolz                     0.12.0             pyhd8ed1ab_0    conda-forge
tornado                   6.1             py310h5764c6d_3    conda-forge
tqdm                      4.64.0          py310h06a4308_0  
traitlets                 5.3.0              pyhd8ed1ab_0    conda-forge
typed-ast                 1.5.3           py310h5764c6d_0    conda-forge
typing                    3.10.0.0           pyhd8ed1ab_0    conda-forge
typing-extensions         4.3.0                hd8ed1ab_0    conda-forge
typing_extensions         4.3.0              pyha770c72_0    conda-forge
tzdata                    2022a                hda174b7_0  
ujson                     5.3.0           py310hd8f1fbe_0    conda-forge
unicodedata2              14.0.0          py310h5764c6d_1    conda-forge
unidecode                 1.3.4              pyhd8ed1ab_0    conda-forge
urllib3                   1.26.9          py310h06a4308_0  
watchdog                  2.1.9           py310hff52083_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.37.1             pyhd3eb1b0_0  
wrapt                     1.14.1          py310h5764c6d_0    conda-forge
wurlitzer                 3.0.2              pyhd8ed1ab_2    conda-forge
xz                        5.2.5                h7f8727e_1  
yaml                      0.2.5                h7b6447c_0  
yaml-cpp                  0.7.0                h295c915_1  
yapf                      0.32.0             pyhd8ed1ab_0    conda-forge
zeromq                    4.3.4                h9c3ff4c_1    conda-forge
zipp                      3.8.0           py310h06a4308_0  
zlib                      1.2.12               h7f8727e_2  
zstd                      1.5.2                ha4553b6_0
```

**Comment by hmaarrfk** [comment]
It seems you have a mixed environment between the default anaconda channel and conda-forge.

That isn't really supported by either anaconda or conda-forge anymkorecan you try creating a new environment? I believe that will pull in things only from conda-forge

If you still get packages from both channels, you should use the strict channel property suggested on conda forges website.

**Comment by jjl2** [comment]
Thank you for the suggestion, @hmaarrfk !

I had an issue when setting up an environment using only the conda-forge channel.  Some dependencies I needed were simply unavailable in conda-forge!  So I added the defaults channel as well.  Perhaps conda-forge was temporarily broken?  I am not sure.

I just created a new virtual environment which used only conda-forge, and this time I was able to install scikit-image and import the segmentation module.  Problem is solved.
